docs/meps/mep-0002.md

```diff
@@ -0,0 +1,364 @@
+- If the reference target is not enumerated, the title will be used. This may be the section header text (including syntax/style) or a figure caption.
+- If the node is not enumerated and does not have a title, the reference label will be used.
+
+If the `text` is included it will be used as is with two additional template values (`{number}` and `{name}`)
+
+- Enumeration (`{number}`)
+
+  - Any reference target that is enumerated can reference that number or string with a `{number}`; the `{number}` is only the enumerator and does not include any preceding text.
+    - For example, if referencing `#my-section` with default text of `Section 2.1.2`, you can change the reference to `See 2.1.2` with `[See {number}](#my-section)`.
+  - If a `{number}` is used and the node is not enumerated, the `{number}` will be replaced by "??" and a warning raised.
+
+- Reference text (`{name}`)
+  - Any node can include the text of a reference including any styles (e.g. Sections are the section title; Figures and Tables are the caption).
+  - If `{name}` is used and the node does not have an explicit name or title, the node reference label will be used.
+
+In both cases, the template can be escaped with a preceding backslash, that is `\{number}` or `\{name}`, and the text will not be replaced.
+
+#### `link`
+
+The links are defined by a scheme, which can be standard protocols (`http:`, `mailto:`).
+Here we propose two new schemes, `path` and `project` which is an extensibility point described by [CommonMark](https://spec.commonmark.org/0.30/#example-598).
+These schemes are used to indicate that the link should be resolved by MyST specific logic, and follows standard [URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) syntax:
+
+```text
+URI = scheme ":" pathname ["?" query] ["#" fragment]
+```
+
+In most cases, as seen in the summary above the scheme is optional and can be inferred safely by the context.
+The exception is when _explicitly_ referring to an external MyST site, Jupyter Book or Sphinx documentation site.
+These URIs can be safely and easily parsed by any common URL parser. For example in Javascript:
+
+```js
+const url = new URL('project:target.md#my-ref');
+url.protocol; // "project:"
+url.pathname; // "target.md"
+url.hash; // "#my-ref"
+```
+
+The following links and references are supported:
+
+| Link Type                | Auto Link                  | Inline                    |
+| :----------------------- | :------------------------- | :------------------------ |
+| External URL             | `<https://example.com>`    | `[](https://example.com)` |
+| Local file download      | `<path:file.txt>`          | `[](file.txt)`            |
+| File download (explicit) | `<path:file.md>`           | `[](path:file.md)`        |
+| Project document         | `<project:file.md>`        | `[](file.md)`             |
```

Comment:
from the table here I found some of the ambiguity confusing. The syntax is fine though, it was just the way it was expressed in the MEP -- but reading on, the Downloads section below does make it clearer what is going on between  `[](path:file.md)` and `[](file.md)` which is (i) download file that is also in the TOC and (ii) link to a file in the TOC(?)

So I think that is fine.